controllers/datadogagent/override/global.go

```diff
@@ -6,16 +6,212 @@
+	if config.Kubelet != nil {
+		var kubeletHostValueFrom *corev1.EnvVarSource
+		if config.Kubelet.Host != nil {
+			kubeletHostValueFrom = config.Kubelet.Host
+		} else {
+			kubeletHostValueFrom = &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: apicommon.FieldPathStatusHostIP,
+				},
+			}
+		}
+		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:      apicommon.DDKubeletHost,
+			ValueFrom: kubeletHostValueFrom,
+		})
+
+		// tlsverify defaults to true in agent code
+		if !*config.Kubelet.TLSVerify {
+			manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+				Name:  apicommon.DDKubeletTLSVerify,
+				Value: "false",
+			})
+		}
+
+		if config.Kubelet.AgentCAPath != "" || config.Kubelet.HostCAPath != "" {
+			manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+				Name:  apicommon.DDKubeletCAPath,
+				Value: getAgentCAPath(config.Kubelet),
+			})
+		}
+
+		if config.Kubelet.HostCAPath != "" {
+			kubeletVol, kubeletVolMount := volume.GetVolumes(apicommon.KubeletCAVolumeName, config.Kubelet.HostCAPath, getAgentCAPath(config.Kubelet), true)
+			manager.VolumeMount().AddVolumeMountToContainers(&kubeletVolMount, getContainerList(manager))
+			manager.Volume().AddVolume(&kubeletVol)
+		}
+	}
+
+	// Path to the docker runtime socket.
+	if *config.DockerSocketPath != "" {
+		dockerMountPath := filepath.Join(apicommon.HostCriSocketPathPrefix, *config.DockerSocketPath)
+		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:  apicommon.DDDockerHost,
+			Value: "unix://" + dockerMountPath,
+		})
+		dockerVol, dockerVolMount := volume.GetVolumes(apicommon.CriSocketVolumeName, *config.DockerSocketPath, dockerMountPath, true)
+		manager.VolumeMount().AddVolumeMountToContainers(&dockerVolMount, getContainerList(manager))
+		manager.Volume().AddVolume(&dockerVol)
+	}
+
+	// Path to the container runtime socket (if different from Docker).
+	if *config.CriSocketPath != "" {
+		criSocketMountPath := filepath.Join(apicommon.HostCriSocketPathPrefix, *config.CriSocketPath)
+		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:  apicommon.DDCriSocketPath,
+			Value: criSocketMountPath,
+		})
+		criVol, criVolMount := volume.GetVolumes(apicommon.CriSocketVolumeName, *config.CriSocketPath, criSocketMountPath, true)
+		manager.VolumeMount().AddVolumeMountToContainers(&criVolMount, getContainerList(manager))
+		manager.Volume().AddVolume(&criVol)
+	}
 
 	return manager.PodTemplateSpec()
 }
+
+func getContainerList(manager feature.PodTemplateManagers) []apicommonv1.AgentContainerName {
+	contList := []apicommonv1.AgentContainerName{}
+	for _, c := range manager.PodTemplateSpec().Spec.InitContainers {
+		contList = append(contList, apicommonv1.AgentContainerName(c.Name))
+	}
+	for _, c := range manager.PodTemplateSpec().Spec.Containers {
+		contList = append(contList, apicommonv1.AgentContainerName(c.Name))
+	}
+	return contList
+}
+
+func getAgentCAPath(kubeletConfig *apicommonv1.KubeletConfig) string {
```

Comment:
i'm thinking it'd be cleaner if we took care of this [logic](https://github.com/DataDog/datadog-operator/blob/f19ecf5d53950e4593b64faab9b8bdcbf66e4008/apis/datadoghq/common/v1/types.go#L62) in the defaulting code as well